core/src/main/java/org/jdbi/v3/GeneratedKeys.java

```diff
@@ -74,4 +77,9 @@
             throw new ResultSetException("Exception thrown while attempting to traverse the result set", e, context);
         }
     }
+
+    @Override
+    public <ContainerType> ContainerType collect(Class<ContainerType> containerType) {
+        return stream().collect(containerFactoryRegistry.createCollectorFor(containerType));
+    }
```

Comment:
I would almost rather just pass in a Collector directly with the fluent API. E.g. `GuavaCollectors.toImmutableList`.